be/src/exec/json_scanner.cpp

```diff
@@ -683,10 +683,11 @@ Status JsonReader::_construct_row(simdjson::ondemand::object* row, Chunk* chunk)
 
 Status JsonReader::_read_file_stream() {
     // TODO: Remove the down_cast, should not rely on the specific implementation.
-    auto pipe = down_cast<StreamLoadPipeInputStream*>(_file->stream().get())->pipe();
+    auto pipe = make_shared<StreamLoadPipeReader>(down_cast<StreamLoadPipeInputStream*>(_file->stream().get())->pipe());
```

Comment:
1. It's better to rename to pipeReader.
2. The StreamLoadPipeReader object will be created even if using compressedReader, right?

Reply:
> 2. The StreamLoadPipeReader object will be created even if using compressedReader, right?

Yes.

Reply:
> 2. The StreamLoadPipeReader object will be created even if using compressedReader, right?

Yes.